src_features/signAuthorizationEIP7702/commands_7702.c

```diff
@@ -0,0 +1,218 @@
+#include "common_ui.h"
+#include "network.h"
+#include "crypto_helpers.h"
+#include "commands_7702.h"
+#include "shared_7702.h"
+#include "rlp_encode.h"
+#include "whitelist_7702.h"
+
+#define MAGIC_7702 5
+
+/**
+ * @brief Read an input parameter passed as a length (1 byte) followed by the parameter
+ * @param [in] dataBuffer buffer containing the length + parameter to read
+ * @param [in] dataLength size of the input buffer
+ * @param [out] output buffer to store the parameter
+ * @param [in,out] output_size size of the output buffer in bytes (input), size of the copied
+ * parameter (output)
+ * @return APDU_NO_RESPONSE if the parameter could be copied, or an error code
+ */
+uint16_t readDataLVParameter(const uint8_t *dataBuffer,
+                             uint8_t dataLength,
+                             uint8_t *output,
+                             uint8_t *outputLength) {
+    uint8_t valueLength;
+    if (dataLength < 1) {
+        return APDU_RESPONSE_INVALID_DATA;
+    }
+    valueLength = *dataBuffer;
+    dataBuffer++;
+    dataLength--;
+    if ((dataLength < valueLength) || (valueLength > *outputLength)) {
+        return APDU_RESPONSE_INVALID_DATA;
+    }
+    memmove(output, dataBuffer, valueLength);
+    *outputLength = valueLength;
+    return APDU_NO_RESPONSE;
+}
+
+/**
+ * @brief Encode a parameter as RLP and hash it using Keccak
+ * @param [in] data parameter to hash
+ * @param [in] dataLength size of the parameter to hash
+ * @param [out] rlpTmp temporary buffer to store the RLP encoded parameter
+ * @param [in] rlpTmpLength size of the temporary buffer to store the RLP encoded parameter
+ * @return APDU_NO_RESPONSE if the parameter could be hashed, or an error code
+ */
+uint16_t hashRLP(const uint8_t *data, uint8_t dataLength, uint8_t *rlpTmp, uint8_t rlpTmpLength) {
+    cx_err_t error = CX_INTERNAL_ERROR;
+    uint8_t hashSize;
+
+    hashSize = rlpEncodeNumber(data, dataLength, rlpTmp, rlpTmpLength);
+    if (hashSize == 0) {
+        return APDU_RESPONSE_UNKNOWN;
+    }
+    CX_CHECK(cx_hash_no_throw((cx_hash_t *) &global_sha3, 0, rlpTmp, hashSize, NULL, 0));
+    return APDU_NO_RESPONSE;
+end:
+    return error;
+}
+
+uint16_t handleSignEIP7702Authorization(const uint8_t *dataBuffer,
+                                        uint8_t dataLength,
+                                        unsigned int *flags) {
+    uint8_t delegate[20];
+    uint8_t chainId[32];
+    uint8_t chainIdLength = sizeof(chainId);
+    uint8_t nonce[8];
+    uint8_t nonceLength = sizeof(nonce);
+    uint8_t rlpDataSize = 0;
+    uint8_t rlpTmp[40];
+    uint8_t hashSize;
+    uint16_t sw;
+    uint64_t chain_id64;
+    cx_err_t error = CX_INTERNAL_ERROR;
+    cx_ecfp_public_key_t publicKey;
+    const char *networkName;
+#ifdef HAVE_EIP7702_WHITELIST
+    const char *delegateName;
+#endif
+
+    // Parse all input parameters
+    // * BIP 32 Path
+    dataBuffer = parseBip32(dataBuffer, &dataLength, &tmpCtx.authSigningContext7702.bip32);
```

Comment:
If the APDU contains a TLV struct, these two functions would need to be used : 
* [tlv_from_apdu](https://github.com/LedgerHQ/app-ethereum/blob/118d237503c0990aa464506234df8b640a8d6708/src/tlv_apdu.h#L9-L12)
* [tlv_parse](https://github.com/LedgerHQ/app-ethereum/blob/118d237503c0990aa464506234df8b640a8d6708/src/tlv.h#L22)

Like it is done for other APDUs containing a TLV struct, [example here](https://github.com/LedgerHQ/app-ethereum/tree/108363fd7f68a7cdf13dcfbfbe0a1a1a2a1cb39f/src_features/provide_proxy_info).